onnxruntime/core/optimizer/qdq_transformer/ensure_unique_dq_for_node_unit.cc

```diff
@@ -29,7 +29,8 @@ constexpr auto* kTransformerName = "EnsureUniqueDQForNodeUnit";
 //                    +-------> DQ' -> Y
 //
 // Note: <DQ_inputs> may be graph inputs/initializers or nodes.
-Status DuplicateDQForOutputEdge(const graph_utils::GraphEdge& original_dq_output_edge, Graph& graph) {
+Status DuplicateDQForOutputEdge(const graph_utils::GraphEdge& original_dq_output_edge, const std::string& dq_domain,
```

Comment:
Could we infer the domain from the data type? 

If we expect the int32 support to be added to onnx, passing through the domain is temporary, so if we infer the domain there's no need to remember that we could go back and remove passing in the domain to clean things up. 

Not sure if there's any scenario where the domain needs to be explicitly set or if the data type info may not be available when needed

Reply:
At certain points in time, the ms_domain version of the DQ operator may differ in capability from the ONNX version (as we experiment with 16bit, 4bit, etc). In my opinion, passing in the domain explicitly makes it easier to ensure that the duplicate DQ operator has the same capabilities as the original.

If we inferred the domain from the zp data type, then we may want to come back and clean-up the inferred domain once the ONNX version catches up with the ms_domain version.  Also, this code would have to keep track of which domain supports the desired zp type. Passing in the domain explicitly removes this burden.

Lastly, I think passing in the domain is the least surprising alternative: the duplicate op has the same type and domain as the original.

Reply:
another alternative is to use the domain directly from the original DQ node at the end of `original_dq_output_edge`. we can probably avoid having a separate parameter.

Reply:
Updated